### PR TITLE
Change CMake behavior for PipeWire, obs-browser and obs-vst if one of them is missing

### PR DESCRIPTION
**Pull request by tytan652**
### Description

If obs-vst, obs-browser or PipeWire/Gio are not present. CMake will emit a fatal error and ask to add them or turn the option off.
All of them are set to ON by default.

If you think that the naming of the commits are bad. Tell me.
If you want me to put better message. Tell me too.

### Motivation and Context

Many Linux packagers doesn't seem to build OBS with obs-vst and obs-browser, or maybe they don't know about them.
So I create this PR, to make the packager really decide to include or not those two.

I also added a fatal error if PipeWire/Gio is not present.

### How Has This Been Tested?

I disabled obs-vst or obs-browser submodules to see if CMake emit the error.

I didn't tested the PipeWire one but since it's based on #4439, it should work. On my Arch, PipeWire is a dependency of GNOME thanks to mutter.

### Types of changes
- CMake changes

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [ ] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by Gol-D-Ace** [comment]
I'm personally supporting this change *cough* Debian package.

File: plugins/linux-capture/CMakeLists.txt [hunk 1]
```diff
@@ -46,8 +46,14 @@ set(linux-capture_LIBRARIES
 
 option(ENABLE_PIPEWIRE "Enable PipeWire support" ON)
 if(ENABLE_PIPEWIRE)
-	find_package(PipeWire REQUIRED)
-	find_package(Gio REQUIRED)
+	find_package(PipeWire QUIET)
+	find_package(Gio QUIET)
+
+	if(NOT PIPEWIRE_FOUND)
+		message(FATAL_ERROR "PipeWire library not found, Please install PipeWire or set ENABLE_PIPEWIRE=OFF")
```
Comment (GeorgesStavracas) [inline plugins/linux-capture/CMakeLists.txt:53]: There's a capitalization issue here and in the message below. Either change the comma to a dot, or de-capitalize `Please` :slightly_smiling_face:

File: plugins/CMakeLists.txt [hunk 2]
```diff
@@ -72,10 +72,13 @@ else()
...
-if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-vst/CMakeLists.txt")
-	add_subdirectory(obs-vst)
-else()
-	message(STATUS "obs-vst submodule not found!  Please fetch/update submodules.  obs-vst plugin disabled.")
+option(BUILD_VST "Build VST plugin" ON)
+if(BUILD_VST)
+	if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-vst/CMakeLists.txt")
+		add_subdirectory(obs-vst)
+	else()
+		message(FATAL_ERROR "obs-vst submodule not found!  Please fetch submodules or set BUILD_VST=OFF.")
```
Comment (GeorgesStavracas) [inline plugins/CMakeLists.txt:80]: Nitpick: two whitespaces after `!`

File: plugins/CMakeLists.txt [hunk 3]
```diff
@@ -60,14 +60,13 @@ elseif("${CMAKE_SYSTEM_NAME}" MATCHES "OpenBSD")
...
-option(BUILD_BROWSER "Build browser plugin" OFF)
+option(BUILD_BROWSER "Build browser plugin" ON)
 if (BUILD_BROWSER)
 	if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-browser/CMakeLists.txt")
 		add_subdirectory(obs-browser)
 		set(BROWSER_AVAILABLE_INTERNAL ON CACHE BOOL "Internal global cmake variable" FORCE)
 	else()
-		message(STATUS "obs-browser submodule not found!  Please fetch submodules.  obs-browser plugin disabled.")
-		set(BROWSER_AVAILABLE_INTERNAL OFF CACHE BOOL "Internal global cmake variable" FORCE)
+		message(FATAL_ERROR "obs-browser submodule not found!  Please fetch submodules or set BUILD_BROWSER=OFF.")
```
Comment (GeorgesStavracas) [inline plugins/CMakeLists.txt:69]: Nitpick: two whitespaces after `!`

**Review by GeorgesStavracas — CHANGES_REQUESTED** [review]

**Comment by tytan652** [comment]
@GeorgesStavracas thank you

**Comment by GeorgesStavracas** [comment]
Er, just noticed I reviewed commits in reverse order. Too used to GitLab, I guess :slightly_smiling_face:

**Comment by tytan652** [comment]
Oh! Flatpak seems to had an issue not related to my commits

**Comment by GeorgesStavracas** [comment]
The Flatpak failure is DockerHub acting up again. A rerun should be enough to fix it.

File: plugins/CMakeLists.txt [hunk 4]
```diff
@@ -60,14 +60,13 @@ elseif("${CMAKE_SYSTEM_NAME}" MATCHES "OpenBSD")
 	add_subdirectory(sndio)
 endif()
 
-option(BUILD_BROWSER "Build browser plugin" OFF)
+option(BUILD_BROWSER "Build browser plugin" ON)
```
Comment (GeorgesStavracas) [inline plugins/CMakeLists.txt:63]: Is this supposed to be here? Enabling the browser is more involving than changing this boolean because it involves setting a path for CEF, which is hard to do automatically. Maybe split the changes to defaults as another pull request so we can discuss it separatedly?

Comment (tytan652) [reply plugins/CMakeLists.txt:63]: It's meant to "force" Linux packagers to see that there is submodules to set up. Because many packagers doesn't seem to know about them. I may need to make a PR for obs-browser to make it emit also a fatal error if CEF is not found.

Comment (GeorgesStavracas) [reply plugins/CMakeLists.txt:63]: Hmmmmm... usually, distros take a tarball generated, for example, by `make distcheck` or `ninja dist`, which should in theory have submodules. I'm not sure what's the release process for OBS, or if Jim rolls out manual tarballs. If packagers are expected to use git tags, than this makes sense. But regardless of that, I guess it's making sure packagers either explicitly disable it, or make sure it actually really works, is a good idea.

Comment (tytan652) [reply plugins/CMakeLists.txt:63]: No submodules in github source code tarball

Comment (tytan652) [reply plugins/CMakeLists.txt:63]: [obs-browser PR](https://github.com/obsproject/obs-browser/pull/285)

File: plugins/linux-capture/CMakeLists.txt [hunk 5]
```diff
@@ -46,8 +46,14 @@ set(linux-capture_LIBRARIES
...
 if(ENABLE_PIPEWIRE)
-	find_package(PipeWire REQUIRED)
-	find_package(Gio REQUIRED)
+	find_package(PipeWire QUIET)
+	find_package(Gio QUIET)
+
+	if(NOT PIPEWIRE_FOUND)
+		message(FATAL_ERROR "PipeWire library not found! Please install PipeWire or set ENABLE_PIPEWIRE=OFF")
+	elseif(NOT GIO_FOUND)
+		message(FATAL_ERROR "Gio library not found! Please install Gio or set ENABLE_PIPEWIRE=OFF")
```
Comment (kkartaltepe) [inline plugins/linux-capture/CMakeLists.txt:55]: Please install gio/glib2 as its usually bundled with glib2 from what i see.

Comment (tytan652) [reply plugins/linux-capture/CMakeLists.txt]: Fixed

**Review by GeorgesStavracas — APPROVED** [review]
